File: 0141-linked-list-cycle/0141-linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        ListNode* temp=head;
        int s=0;
        if(head==nullptr){
            return head;
        }
        while(temp->next!=nullptr){
            temp=temp->next;
            s++;
            if(s>10000000){
                return true;
            }
        }
        return false;
    }
};

File: 0141-linked-list-cycle/0141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode rabbit=head;
        ListNode tor=head;
        while(rabbit!=null && rabbit.next!=null){
            rabbit=rabbit.next.next;
            tor=tor.next;
            if(tor==rabbit) 
                return true;
        }
        return false;
    }
}